question19/js/slideshow.js

```diff
@@ -0,0 +1,183 @@
+    if (!response.ok) {
+      console.error(`${response.status}:${response.statusText}`);
+    }
+    const json = await response.json();
+    return json.data;
+  } catch (error) {
+    console.error(error);
+  } finally {
+    removeLoadImg();
+  }
+};
+
+const init = async () => {
+  const imgData = await callApi();
+  if (imgData) {
+    renderImgUiElement(imgData);
+  } else {
+    imgListsWrapper.textContent = "データの表示に失敗しました";
+    console.error("データの表示に失敗しました");
+  }
+};
+init();
+
+const renderImgUiElement = (imgData) => {
+  createBtnElements(imgData);
+  renderPageNumElement(imgData);
+  renderPagenation(imgData);
+  clickedPagenation(imgData);
+  autoImgSwitch(imgData);
+  imgLists.appendChild(getFragmentImglists(imgData));
+};
+
+const renderPageNumElement = (imgData) => {
+  const numberOfPage = createElementWithClassName("p", "pageNumber");
+  imgListsWrapper.appendChild(numberOfPage);
+  numberOfPage.id = "js-number";
+  numberOfPage.textContent = `1 / ${imgData.length}`;
+};
+
+const getFragmentImglists = (imgData) => {
+  const fragmentImglists = document.createDocumentFragment();
+  for (let i = 0; i < imgData.length; i++) {
+    const imgList = createElementWithClassName("li", "imgList");
+    const img = createElementWithClassName("img", `img_0${[i + 1]}`);
+    imgList.dataset.index = i;
+    img.src = imgData[i].img;
+    img.alt = imgData[i].description;
+    i === 0 && imgList.classList.add("is-show");
+    fragmentImglists.appendChild(imgList).appendChild(img);
+  }
+  return fragmentImglists;
+};
+
+const imgNum = { count: 0 };
+const imgList = document.getElementsByClassName("imgList");
+
+const createBtnElements = (imgData) => {
+  const btnDirections = ["next", "prev"];
+
+  btnDirections.forEach((btnDirections) => {
+    const btn = document.createElement("button");
+    btn.classList.add(`${btnDirections}`);
+    btn.id = btnDirections === "prev" ? "js-prevbtn" : "js-nextbtn";
+    btn.textContent = btnDirections === "prev" ? "◀︎" : "▶︎";
+    imgListsWrapper.appendChild(btn);
+    btn.disabled = btnDirections === "prev";
+
+    btn.addEventListener("click", function () {
+      btnDirections === "prev" ? (imgNum.count -= 1) : (imgNum.count += 1);
+      document.querySelector(".is-show").classList.remove("is-show");
+      imgList[imgNum.count].classList.add("is-show");
+      switchPagenation(imgNum.count);
+      switchDisableForBtn(imgData);
+      setNumberOfPage(imgData);
+      resetAutoPlay(imgData);
+    });
+  });
+};
+
+const setNumberOfPage = (imgData) => {
+  document.getElementById("js-number").textContent = `${imgNum.count + 1} / ${
+    imgData.length
+  }`;
+};
+
+const switchDisableForBtn = (imgData) => {
+  const prevBtnElement = document.getElementById("js-prevbtn");
+  const nxtBtnElement = document.getElementById("js-nextbtn");
+  const lengthImg = imgData.length;
+
+  prevBtnElement.disabled = imgNum.count === 0;
+  nxtBtnElement.disabled = imgNum.count === lengthImg - 1;
+};
```

Comment:
nits
I thought it would be easier to understand what the conditional branch expression does if 0 and lengthImg-1 are variables.
```suggestion
const switchDisableForBtn = (imgData) => {
  const prevBtnElement = document.getElementById("js-prevbtn");
  const nxtBtnElement = document.getElementById("js-nextbtn");
  const firstNum = 0;
  const lastNum = imgData.length -1;

  prevBtnElement.disabled = imgNum.count === firstNum;
  nxtBtnElement.disabled = imgNum.count === lastNum;
};
```

Reply:
Thank you for your review!! You are right! I thought it was very readable!! Thank you for pointing it out!
fixed commit : https://github.com/csakyo/homework/pull/33/commits/19a7d04788ede695449eca39dec24e691eca6c1b